### ISSUE TITLE
C# completion improperly escapes \ within @ strings

### ISSUE THREAD

**Issue by jonpryor** [mono/monodevelop#5489]
C# @-strings don't require that `\` be escaped; that's the point.  This makes @-strings ideal for regular expressions (and Windows filesystem paths).

Unfortunately, the Visual Studio for Mac IDE is weird here.

Start with the beginnings of a regex:

```csharp
var r = new Regex (@"");
```

All is fine.

Now, *edit* the string of the regex and *type* `\d+`:

After the initial `\d`, things look quite nice; we get code completion on a regex!

```csharp
var r = new Regex (@"\d");
// ... \d - Digit character
```

However, when I type the `+`, the `\` is doubled:

```csharp
var r = new Regex (@"\\d+");
```

This is *wrong*, so now I have to manually fixup what the IDE "helpfully" inserted.

Version details:

```
Visual Studio Enterprise 2017 for Mac
Version 7.5.3 (build 7)
Installation UUID: f20c27b8-9b9d-492a-9391-8f510637a429
Runtime:
	Mono 5.10.1.57 (2017-12/ea8a24b1bbf) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.1.178 (master / eeaeb7e6)

	Package version: 510010057

NuGet
Version: 4.3.1.4445

.NET Core
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.0.5
	2.0.0
	1.1.1
	1.0.4
	1.0.0-rc2-3002702
SDK: /usr/local/share/dotnet/sdk/2.1.4/Sdks
SDK Versions:
	2.1.4
	2.0.0
	1.0.3
	1.0.0-preview1-002702
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.10.1/lib/mono/msbuild/15.0/bin/Sdks

Xamarin.Profiler
Version: 1.6.2
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Apple Developer Tools
Xcode 9.4.1 (14161)
Build 9F2000

Xamarin.Mac
Version: 4.4.1.193 (Visual Studio Enterprise)

Xamarin.iOS
Version: 11.12.0.4 (Visual Studio Enterprise)
Hash: 64fece5f
Branch: d15-7
Build date: 2018-05-29 20:00:44-0400

Xamarin.Android
Version: 8.3.3.2 (Visual Studio Enterprise)
Android SDK: /Users/jon/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		2.3   (API level 10)
		4.0.3 (API level 15)
		4.4   (API level 19)
		5.0   (API level 21)
		6.0   (API level 23)
		7.1   (API level 25)
		8.0   (API level 26)
		8.1   (API level 27)

SDK Tools Version: 25.1.2
SDK Platform Tools Version: 24.0.0
SDK Build Tools Version: 23.0.2

Java SDK: /usr
java version "1.8.0_77"
Java(TM) SE Runtime Environment (build 1.8.0_77-b03)
Java HotSpot(TM) 64-Bit Server VM (build 25.77-b03, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Xamarin Inspector
Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

Build Information
Release ID: 705030007
Git revision: 13cecd02aceddf29a1ed57b86f81c02994df1483
Build date: 2018-06-14 15:48:08-04
Xamarin addins: 7065de97cf22c9038fdc39dd627f2c30790fd8af
Build lane: monodevelop-lion-d15-7

Operating System
Mac OS X 10.13.5
Darwin 17.6.0 Darwin Kernel Version 17.6.0
    Tue May  8 15:22:16 PDT 2018
    root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64

Enabled user installed addins
Internet of Things (IoT) development (Preview) 7.5
```

> VS bug [#653156](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/653156)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.741 (master)

Pull Request #5496 merged by: Mike Krüger
Author: mono
Commit: 49b5da0585b5f0c35697e0241ba9f867218ea918 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5496 merged by: Mike Kr??ger
Author: mono
Commit: 49b5da0585b5f0c35697e0241ba9f867218ea918 (xamarin/monodevelop)
